packages/gestalt/src/SheetConfirmationPopover.js

```diff
@@ -0,0 +1,161 @@
+// @flow strict
+
+import { type Node, useEffect, useRef } from 'react';
+import { useAnimation } from './AnimationContext.js';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Popover from './Popover.js';
+import Button from './Button.js';
+import Text from './Text.js';
+import TrapFocusBehavior from './behaviors/TrapFocusBehavior.js';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider.js';
+import { ESCAPE } from './keyCodes.js';
+
+type Props = {|
+  onDismiss: () => void,
+  anchor: ?HTMLElement,
+  dismissConfirmation?:
+    | boolean
+    | {|
+        message?: string,
+        subtext?: string,
+        primaryAction?: {|
+          accessibilityLabel?: string,
+          text?: string,
+          onClick?: ({|
+            event:
+              | SyntheticMouseEvent<HTMLButtonElement>
+              | SyntheticMouseEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLButtonElement>,
+          |}) => void,
+        |},
+        secondaryAction?: {|
+          accessibilityLabel?: string,
+          text?: string,
+          onClick?: ({|
+            event:
+              | SyntheticMouseEvent<HTMLButtonElement>
+              | SyntheticMouseEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLAnchorElement>
+              | SyntheticKeyboardEvent<HTMLButtonElement>,
+          |}) => void,
+        |},
+      |},
+|};
+
+export default function SheetConfirmationPopover({
+  anchor,
+  dismissConfirmation,
+  onDismiss,
+}: Props): Node {
+  const { onAnimatedDismiss } = useAnimation();
+
+  const {
+    dismissConfirmationMessage: dismissConfirmationMessageDefault,
+    dismissConfirmationSubtext: dismissConfirmationSubtextDefault,
+    dismissConfirmationPrimaryActionText: dismissConfirmationPrimaryActionTextDefault,
+    dismissConfirmationSecondaryActionText: dismissConfirmationSecondaryActionTextDefault,
+    dismissConfirmationPrimaryActionTextLabel: dismissConfirmationPrimaryActionTextLabelDefault,
+    dismissConfirmationSecondaryActionTextLabel: dismissConfirmationSecondaryActionTextLabelDefault,
+  } = useDefaultLabelContext('Sheet');
+
+  const confirmationButtonRef = useRef();
+
+  useEffect(() => {
+    if (confirmationButtonRef.current) {
+      confirmationButtonRef.current.focus();
+    }
+  }, [confirmationButtonRef]);
+
+  // Handle onDismiss triggering from ESC keyup event
+  useEffect(() => {
+    function handleKeyDown(event: {| keyCode: number, stopPropagation: () => void |}) {
```

Comment:
Do we need to add the `event` type here? Shouldn't Flow be able to infer that?

Reply:
DONE